From 25c5c1e1e2e3b4aef7f84c9506fffdcff6716b83 Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Thu, 30 May 2024 10:12:20 +0200
Subject: [PATCH 0856/1695] dts: stm32: add IPCC1 interrupts to stm32mp2x
 platforms

The secure IPCC1 interrupt RXO is required for
mailbox support. TXF interrupt is not configured since
TXF interrupt mask cannot written by secure in an
atomic way to prevent a concurrent access done between
secure and non secure.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 352d1aebea3bec1d14d5f384c76dc95fe80790f4 dts: stm32: add IPCC1 interrupts to stm32mp2x platforms]
Change-Id: I8bab5e4a7c4aaff0604d20dd3898deccfbd422ff
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/379904
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp251.dtsi | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 6949a6ee0..4d129df01 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -201,6 +201,8 @@
 		ipcc1: mailbox@40490000 {
 			compatible = "st,stm32mp25-ipcc";
 			reg = <0x40490000 0x400>;
+			interrupts = <GIC_SPI 173 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "rx";
 			clocks = <&rcc CK_BUS_IPCC1>;
 		};
 
-- 
2.47.2

